ENH: Fixed PercentFormatter usage with latex

madphysicist opened the pull request:
This PR addresses a nagging issue that I have had with PercentFormatter, namely that it does not handle LaTeX labels correctly unless preconfigured to do so. The proposed solution allows the labels to be generated correctly with the same formatter even if `rcParams['text.usetex']` changes between calls.

I also added it to the big formatter example and modified one of the existing pylab examples where it clearly belongs.

 This PR also includes the of IDs to `test_percentformatter` that I made in #7482.

madphysicist commented:
@story645: given your concern with `PercentFormatter`, I think you might be a good person to have reviewing this.

NelleV commented on lib/matplotlib/ticker.py:
```diff
@@ -1353,13 +1356,30 @@ def format_pct(self, x, display_range):
     def convert_to_pct(self, x):
         return 100.0 * (x / self.xmax)
 
+    def get_symbol(self):
```
Can you make this function private? If possible, we try to avoid adding more elements to the public API that aren't absolutely necessary.

madphysicist replied:
How about making `self.symbol` into a property since it's already public and removing the need for this method entirely?

NelleV replied:
sounds good.

madphysicist commented on lib/matplotlib/tests/test_ticker.py:
```diff
@@ -536,12 +536,66 @@ def test_formatstrformatter():
 
 
 @pytest.mark.parametrize('xmax, decimals, symbol, x, display_range, expected',
-                         percentformatter_test_cases)
```
Accidentally copied instead of cutting, then forgot because the tests passed. Will fix.

story645 replied:
same comment as on other PR, maybe keep the testcase and id outside of the decorator and just refer to 'em?

story645 commented on examples/pylab_examples/histogram_percent_demo.py:
```diff
@@ -1,30 +1,15 @@
-
 # Set the formatter
-plt.gca().yaxis.set_major_formatter(formatter)
+plt.gca().yaxis.set_major_formatter(PercentFormatter(xmax=1))
```
I know this is beyond the scope of this doc, but is there anyway to do this same thing without a `plt.gca().yaxis`? Also comment should possibly explain what xmax means here

madphysicist replied:
This is not out of scope at all if you are OK with an extra call to `plt.subplots()`.

madphysicist commented:
I have responded to all the review comments. I will make the lists of test parameters into separate references in #7482

madphysicist commented:
I am pretty sure that the Appveyor failure is not something I did, but I could be very wrong about that.

story645 commented:
You could click through and see if the error has anything to do with you,  but travis and appveyor have also been messed around with a lot lately.

madphysicist commented:
Well, #7528 started getting the same error after I rebased, so I am pretty sure it's something on master. [Blame](https://github.com/matplotlib/matplotlib/blame/master/lib/matplotlib/tests/test_colors.py#L618) says @QuLogic is the last person to modify that particular test about 10 days ago, but it may not be the test itself that is causing the failure.

QuLogic commented on examples/ticks_and_spines/tick-formatters.py:
```diff
@@ -21,8 +21,8 @@ def setup(ax):
 
-plt.figure(figsize=(8, 5))
-n = 6
+plt.figure(figsize=(9, 5))
```
You've changed the width, not the height.

QuLogic commented on lib/matplotlib/tests/test_ticker.py:
```diff
@@ -511,35 +511,66 @@ def test_formatstrformatter():
     assert '002-01' == tmp_form(2, 1)
 
 
-percentformatter_test_cases = (
```
I thought @story645 suggested keeping this.

story645 replied:
Got sorted out in #7993

QuLogic replied:
They'll conflict then; which should be merged first?

madphysicist replied:
#7993 please. I'd like to rebase all my open PRs onto it. It's ready to go, just waiting for Mac test to run.

madphysicist commented:
@QuLogic. I fixed the size issue. In addition to #7993, the order of this PR with respect to #7482 matters as well since they both add examples to the big formatter demo. I updated the same resizing issue in #7482 as well. #7482 should come after this PR. Just out of curiosity, is there some mechanism in GitHub for specifying the order in which PRs should be merged?

madphysicist commented:
Fixed PEP8 errors, should be ready to merge when tests are done. This PR needs to come before #7482 because of the conflicts between them.

madphysicist commented:
I suspect that the break may not be related to my tests...

madphysicist commented:
ready to merge with one more approval.

QuLogic commented on examples/pylab_examples/histogram_percent_demo.py:
```diff
@@ -1,30 +1,19 @@
-# default labels by 100, making them all percentages
-formatter = FuncFormatter(to_percent)
+# Make a normed histogram. It'll be multiplied by 100 later.
+plt.hist(x, bins=50, normed=True, axes=ax)
```
Doesn't `ax.hist` work here?

story645 replied:
it's a pylab example so I think it has to be plt.hist

QuLogic replied:
@madphysicist already changed it to `fig, ax`, so I don't see why.

madphysicist replied:
Technically, `plt.*` is already not pylab, so I will go ahead and make the change. I am pretty sure that there are plenty of pylab examples that demo the OO API.

jenshnielsen replied:
I agree @madphysicist a good deal of time was spent removing the use of pylab from all the examples last year. In my opinion most of the pylab examples should be moved to other folders with more meaningful names as they are cleaned up and improved (But I am out of touch on what the current plan is for restructuring the examples)

QuLogic commented on lib/matplotlib/tests/test_ticker.py:
```diff
@@ -565,33 +592,24 @@ class TestPercentFormatter(object):
             percent_data, ids=percent_ids)
-    def test_percentformatter(self, xmax, decimals, symbol,
+    def test_basic(self, xmax, decimals, symbol,
                               x, display_range, expected):
```
Not aligned any more.
